tagging_pipeline/transform.py

```diff
@@ -0,0 +1,109 @@
+"""Makes POST requests to the openai API to find three main topics relating to the article title"""
+
+import os
+import re
+
+import pandas as pd
+from dotenv import load_dotenv
+from nltk.corpus import wordnet, stopwords
+from gensim.models import word2vec
+import spacy
+import numpy as np
+
+
+
+
+def get_story_topics(response) -> list:
+    """Extracts dictionaries containing each story id and associated topics from the openai response"""
+
+    all_topics = response['choices'][0]['message']['content']
+    all_dicts = re.findall('\{(.*?)\}', all_topics)
+    cleaned_topics = []
+
+    for topic in all_dicts:
+        try:
+            topic_split = topic.split(':')
+            cleaned_topics.append({int(topic_split[0]): eval(topic_split[1])})
+        except:
+            continue
+    return cleaned_topics
+
+
+def read_topics_csv():
+    """Returns most recent csv file in topics folder"""
+    csv_files = os.listdir('topics')
+    topics_df = pd.read_csv(f'topics/{sorted(csv_files)[0]}')
+    return topics_df
+
+
+def get_topics_list(topics_df):
+    topics_list = []
+    topics = topics_df[['topic_one', 'topic_two',
+                        'topic_three']].values.tolist()
+    for sublist in topics:
+        for topic in sublist:
+            topic = re.sub("[^a-zA-Z]", " ", topic, re.I)
+            topics_list.append(topic.lower().strip())
+    return topics_list
+
+
+def get_relevant_topics_list(topics_list):
+    stops = set(stopwords.words("english"))
+    relevant_topics = [
+        word for word in topics_list if not word in stops]
+    return relevant_topics
+
+
+def find_synset_for_topic(topic):
+    synonyms = set()
+    for syn in wordnet.synsets(topic):
+        for lemma in syn.lemmas():
+            synonyms.add(lemma.name())
+    return synonyms
+
+
+def find_syn_testing(response_topics):
+    for topic in response_topics:
+        print(find_synset_for_topic(topic))
+
+
+def match_similar_topics(relevant_topics):
+    nlp_topics = [nlp(topic) for topic in relevant_topics]
+    grouped = {}
+    for i in nlp_topics:
+        if i not in grouped:
+            for j in nlp_topics:
+                if j not in grouped:
+                    if i.similarity(j) >= 0.7 and not i.similarity(j) == 1:
+                        grouped[i] = j
+    return grouped
+
+
+def matching_tests(response_topics, general_topics):
+    general_topics = [nlp(topic) for topic in general_topics]
+    response_topics = [nlp(topic) for topic in response_topics]
+    grouped = {}
+    for topic in response_topics:
+        for general in general_topics:
+            if topic.similarity(general) >= 4:
+                grouped[general] = topic
+    return grouped
+
+
+def loop_relevant_topics(relevant_topics):
+    for topic in relevant_topics:
+        return find_synset_for_topic(topic)
+
```

Comment:
I'm not sure a loop needs to be used here. If you are returning after the first time the loop is run this could be `return find_synset_for_topic(relevant_topics[0])`